src/registrar/management/commands/clean_tables.py

```diff
@@ -56,14 +56,24 @@ def handle(self, **options):
             self.clean_table(table_name)
 
     def clean_table(self, table_name):
-        """Delete all rows in the given table"""
+        """Delete all rows in the given table.
+
+        Delete in batches to be able to handle large tables"""
         try:
             # Get the model class dynamically
             model = apps.get_model("registrar", table_name)
-            # Use a transaction to ensure database integrity
-            with transaction.atomic():
-                model.objects.all().delete()
-            logger.info(f"Successfully cleaned table {table_name}")
+            BATCH_SIZE = 1000
```

Comment:
Suggestion for code below -- my reasoning is that I'm not the biggest fan of `while True:` as we could have an infinite loop:

```
try:
        # Get the model class dynamically
        model = apps.get_model("registrar", table_name)
        BATCH_SIZE = 1000
        total_deleted = 0
        
        # Get initial batch of primary keys
        pks = list(model.objects.values_list("pk", flat=True)[:BATCH_SIZE])
        
        # Loop until no more primary keys are returned
        while pks:
            # Use a transaction to ensure database integrity
            with transaction.atomic():
                deleted, _ = model.objects.filter(pk__in=pks).delete()
                total_deleted += deleted
            logger.debug(f"Deleted {deleted} {table_name}s, total deleted: {total_deleted}")
            
            # Get next batch of primary keys
            pks = list(model.objects.values_list("pk", flat=True)[:BATCH_SIZE])
        
        logger.info(f"Successfully cleaned table {table_name}, deleted {total_deleted} rows")
    except LookupError:
        logger.error(f"Model for table {table_name} not found.")
    except Exception as e:
        logger.error(f"Error cleaning table {table_name}: {e}")
```

Reply:
++ for avoiding while true, though wouldn't block on it fully

Reply:
Ok, I made this change